### PR TITLE
Basic Redis implementation & Google Oauth

### PR DESCRIPTION
**Pull request by natebrennand**
- sets up google oauth
- sets up redis db conection pool

### CODE REVIEW

**Comment by brishin** [comment]
Reminding myself to look at this. Btw, don't have merge rights.

**Comment by natebrennand** [comment]
google credentials are in secrets if you want to test locally

**Comment by natebrennand** [comment]
@brishin, can you check this out?

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ env:
     - PORT=5000
     - HOST=0.0.0.0
     - DEBUG=TRUE
+    - SECRET_KEY=abc123
```
Comment (brishin) [inline .travis.yml:11]: Nit: make secret key self documenting that it is purposely not shared.

Comment (natebrennand) [inline .travis.yml:11]: We can add comments to the config file but it doesn't really matter for the test build. You have to set a `secret_key` to be able to use `flask.session`.

The key in production will be randomly generated. Also, this is only to let people find their tokens easily if they've already gone through OAuth (along the same lines as the top right token access in the [current docs](http://data.adicu.com/docs/Documentation)). The session won't have anything to do with rate-limiting.

**Pull request merged**